### ISSUE TITLE
Comma dangle bug

### ISSUE THREAD

**Issue by jpolo** [eslint/eslint#7614]
While upgrading eslint from 3.7.1 to latest version (3.10.2), the rule "comma-dangle" seems broken.
It reports errors when it should not.

```javascript
    ...
    "babel-core": "6.18.2",
    "babel-eslint": "7.1.0",
    "babel-loader": "6.2.7",
    "babel-plugin-transform-class-properties": "6.18.0",
    "babel-plugin-transform-flow-strip-types": "6.18.0",
    "babel-plugin-transform-object-rest-spread": "6.16.0",
    "babel-preset-es2015": "6.18.0",
    "babel-preset-react": "6.16.0",
    "babel-preset-react-hmre": "1.1.1",
    ...
    "eslint": "3.10.2",
    "eslint-loader": "1.6.1",
    "eslint-plugin-flowtype": "2.25.0",
    "eslint-plugin-import": "2.2.0",
    "eslint-plugin-react": "6.7.1",
```

My config
```javascript
'rules': {
   ...
    'comma-dangle': ['error', 'always-multiline'],
   ...
```

My output
```
53:15  error  Unexpected trailing comma  comma-dangle
```

The JS code:
```javascript
...
const {
      appId,
      appName,
      className,
      country,
      locale,
      style,
      ...other, //This is the line 53
 } = this.props;
```

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report.

However, this is working as intended; your code has invalid syntax. Trailing commas are not allowed after rest properties.

For more information, see https://github.com/eslint/eslint/issues/7422, https://github.com/eslint/eslint/issues/7463, https://github.com/eslint/eslint/issues/7502, and https://github.com/eslint/eslint/issues/7297.

**Comment by jpolo** [comment]
Thank you very much for the enlightment !
